### PR TITLE
Optimize the method that gets the downloads count for a given download

### PR DESCRIPTION
**Pull request by Konamiman**
### Changes proposed in this Pull Request:

In order to retrieve the count of downloads performed for a given file in a given order there's the `WC_Customer_Download::get_download_count` method. This method was very inefficient as it was using `WC_Customer_Download_Log_Data_Store::get_download_logs_for_permission` and then locally counting the rows retrieved.

This pull request:

1. Extends `WC_Customer_Download_Log_Data_Store::get_download_count` to accept the value `count` in `$args['returns']`, this will transform the `SELECT *` query into a `SELECT COUNT` query and the method will simply return the retrieved number.
2. Adds a new `WC_Customer_Download_Log_Data_Store::get_download_logs_count_for_permission` method that uses the above change.
3. Modifies `WC_Customer_Download::get_download_count` to use the new method.

### How to test the changes in this Pull Request:

1. Create a downloadable product with one file.
2. As a customer, make an order for the product.
3. Mark the order as completed.
4. As the customer, download the file a few times.
5. Run the following in CLI (setting the proper order id) and verify that you get the proper count of downloads:

```
wp eval '$download_id=WC_Data_Store::load("customer-download")->get_downloads(["order_id"=>ORDER_ID_HERE, "return"=>"ids"])[0]; echo (new WC_Customer_Download($download_id))->get_download_count();'
```

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/data-stores/class-wc-customer-download-log-data-store.php [hunk 1]
```diff
@@ -171,9 +171,11 @@ public function get_download_logs( $args = array() ) {
 			)
 		);
 
+		$is_count = 'count' === $args['return'];
+
 		$query   = array();
 		$table   = $wpdb->prefix . self::get_table_name();
-		$query[] = "SELECT * FROM {$table} WHERE 1=1";
+		$query[] = 'SELECT ' . ( $is_count ? 'COUNT(1)' : '*' ) . " FROM {$table} WHERE 1=1";
```
Comment (naman03malhotra) [inline plugins/woocommerce/includes/data-stores/class-wc-customer-download-log-data-store.php:178]: Probably we should cover this change with unit test?

**Review by naman03malhotra — COMMENTED** [review]
image

Thanks for working on it, works well for me. Left a comment about unit tests.

File: plugins/woocommerce/includes/data-stores/class-wc-customer-download-log-data-store.php (hunk 1 above)
Comment (Konamiman) [reply plugins/woocommerce/includes/data-stores/class-wc-customer-download-log-data-store.php:178]: The thing is that currently there aren't unit tests at all for the customer download logs data store class. Adding them would probably require extensive refactoring to the class, so if we do it I'd rather open a separate PR (or at least, an issue to keep track of the task) instead.

**Review by naman03malhotra — APPROVED** [review]
>so if we do it I'd rather open a separate PR (or at least, an issue to keep track of the task) instead.

okay makes sense, approving this PR, thank you for working on it.

**Pull request merged**